crates/cast/bin/cmd/wallet/vanity.rs

```diff
@@ -327,6 +327,10 @@ impl VanityMatcher for RegexMatcher {
 }
 
 fn parse_pattern(pattern: &str, is_start: bool) -> Result<Either<Vec<u8>, Regex>> {
+    if pattern.chars().any(|c| !c.is_ascii_hexdigit()) {
```

Comment:
this is not entirely right because as you can see below if the hex decode fails then it's used as a regex

Reply:
this is correct but a regex of non hex can never be found right?

like I could never find a prefix of `0xZ...`

Reply:
yes, but then also modifiers like {} ?*+ etc cant be used